composition_stats/__init__.py

```diff
@@ -464,9 +467,15 @@ def clr(mat):
     array([-0.79451346,  0.30409883,  0.5917809 , -0.10136628])
 
     """
-    lmat = np.log(mat)
-    gm = lmat.mean(axis=-1, keepdims=True)
-    return (lmat - gm).squeeze()
+    if ignore_zero:
+        msk = mat.astype(bool)
+        lmat = np.log(mat, where=msk)
+        gm = lmat.mean(axis=-1, keepdims=True, where=msk)
+        return np.where(msk, (lmat - gm).squeeze(), 0.)
+    else:
+        lmat = np.log(mat)
+        gm = lmat.mean(axis=-1, keepdims=True)
+        return (lmat - gm).squeeze()
```

Comment:
I haven't tested it, but maybe it's more efficient to use the ufunc's `where=` parameter here?

```suggestion
    if ignore_zero:
        where = (mat != 0)
        out = np.zeros_like(mat, dtype=float)
    else:
        where = True
        out = None
    lmat = np.log(mat, where=where, out=out)
    gm = lmat.mean(axis=-1, keepdims=True, where=where)
    return np.subtract(lmat, gm, where=where, out=lmat).squeeze()
```